ir/memory.cpp

```diff
@@ -1021,11 +1021,25 @@ void Memory::mk_nonlocal_val_axioms(bool skip_consts) {
     Byte byte(*this, non_local_block_val[i].val.load(offset));
     Pointer loadedptr = byte.ptr();
     expr bid = loadedptr.getShortBid();
+
+    unsigned upperbid = numNonlocals() - 1;
+    expr bid_cond(true);
+    if (has_fncall) {
+      // initial memory cannot contain a pointer to fncall mem block.
+      if (state->isSource()) {
```

Comment:
what if it's target and there are no target-only glb vars? Shouldn't we decrease upperbid as well?
e.g.
```c++
if (is_fncall_mem(upperbid)) {
  --upperbid;
} else if (...) { }
```

Reply:
yes, right. I fixed this